File: BE-Task_1-Php/internal/repository/memory/agent_repository.php
<?php

declare(strict_types=1);

namespace App\Internal\Repository\Memory;

use App\Internal\Domain\Entities\Agent;
use App\Internal\Repository\Interfaces\AgentRepositoryInterface;

final readonly class AgentRepository implements AgentRepositoryInterface
{
    public function __construct(private Store $store)
    {
    }

    public function findById(string $agentId): ?Agent
    {
        return $this->store->agents[$agentId] ?? null;
    }

    public function listAll(): array
    {
        return array_values($this->store->agents);
    }
}

File: BE-Task_1-Ruby/internal/repository/memory/agent_repository.rb
module RefundBackend
  module Repository
    module Memory
      class AgentRepository < RefundBackend::Repository::Interfaces::AgentRepository
        def initialize(store)
          @store = store
        end

        def find_by_id(agent_id)
          @store.fetch_agent(agent_id)
        end

        def list_all
          @store.list_agents
        end
      end
    end
  end
end
